src/scales/scale.category.js

```diff
@@ -49,10 +49,21 @@ module.exports = Scale.extend({
 
 	getLabelForIndex: function(index, datasetIndex) {
 		var me = this;
-		var data = me.chart.data;
+		var chart = me.chart;
+		var data = chart.data;
 		var isHorizontal = me.isHorizontal();
 
-		if (data.yLabels && !isHorizontal) {
+		var ds = chart.getDatasetMeta(datasetIndex).controller;
```

Comment:
I'm not sure what `ds` stands for. Maybe we can call it `controller` or `ctrl` or something along those lines?